File: cityflagtestios/cityflagtestios/AppDetail.swift

//
//  Appdetail.swift
//  cityflagtestios
//
//  Created by Marco Antonio Martínez Gutiérrez on 1/30/20.
//  Copyright © 2020 Marco Antonio Martínez Gutiérrez. All rights reserved.
//


struct AppDetail {
    let name:String
    let images:[Image]
    let summary : String
    let price : Price
    let contentType : ContentType
    let rights : String?
    let title : String
    let links : [Link]
    let id : Id
    let artist : Artist
    let category : Category
    let releaseDate : ReleaseDate
}

struct Image {
    let url:String
    let height:Int
}

struct Price {
    let amount:Double
    let currency:String
}

struct ContentType {
    let term:String
    let label:String?
}

struct Link {
    let duration:String?
    let term:String?
    let title:String?
    let rel:String?
    let type:String?
    let href:String?
    let assetType:String?
}

struct Id{
    let id:String
    let bundleId:String
    let url:String
}

struct Artist{
    let name:String?
    let href:String?
}

struct Category{
    let id:String?
    let term:String
    let label:String
    let scheme:String
}

struct ReleaseDate{
    let label:String
    let date:String
}

File: cityflagtestandroid/app/src/main/java/com/jamadev/cityflagtest/applications/model/AppDetail.kt
package com.jamadev.cityflagtest.applications.model

data class AppDetail(
    val name : String,
    val images : List<Image>,
    val summary : String,
    val price : Price,
    val contentType : ContentType,
    val rights : String?,
    val title : String?,
    val links : List<Link>,
    val id : Id,
    val artist: Artist,
    val category: Category,
    val releaseDate : ReleaseDate
)

data class Image(
    val url : String,
    val height : Int
)

data class Price(
    val amount:Double,
    val currency: String
)

data class ContentType(
    val term : String?,
    val label: String?
)

data class Link(
    val duration : String?,
    val title: String?,
    val rel: String?,
    val type: String?,
    val href: String?,
    val assetType: String?
)

data class Id(
    val id : String,
    val bundleId : String,
    val url : String
)

data class Artist(
    val name : String?,
    val href: String?
)

data class Category(
    val id : String?,
    val term: String,
    val label: String,
    val scheme: String
)

data class ReleaseDate(
    val label : String,
    val date: String
)

